core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -330,13 +342,19 @@ public GetResult get(Get get, Function<String, Searcher> searcherFactory) throws
                         return new GetResult(true, versionValue.version(), op.getSource());
                     }
                 }
-            }
 
-            // no version, get the version from the index, we know that we refresh on flush
-            return getFromSearcher(get, searcherFactory);
+                return getFromSearcher(get, this::acquireInternalSearcher);
```

Comment:
this seems to bypass the IndexSearcherWrapper defined in IndexShard (and passed via the searchFactory)